aiven_db_migrate/migrate/pgutils.py

```diff
@@ -26,7 +29,8 @@ def find_pgbin_dir(pgversion: str, *, max_pgversion: Optional[str] = None, usr_d
     # We need a special case for versions 9:  The sub-version counts as a major version too
     max_parts = 2 if min_version[0] == 9 else 1
     candidates = []
-    search_scopes = [(usr_dir, r"pgsql-([0-9]+(\.[0-9]+)*)"), (usr_dir / "lib/postgresql", r"([0-9]+(\.[0-9]+)*)")]
+    search_scopes = [(usr_dir, r"pgsql-([0-9]+(\.[0-9]+)*)"), (usr_dir / "lib/postgresql", r"([0-9]+(\.[0-9]+)*)"),
+                     (usr_dir, r"([0-9]+(\.[0-9]+)*)")]
```

Comment:
The newly-added scope dominates over the previous one if I'm not mistaken, and it's quite broad. It will match *ANY* directory with a number in usr_dir, right? It could be any dir with a version number and a bin directory, since our code doesn't check for the presence of actual pg binaries.

I don't have a mac with homebrew right now to test this; but I suppose we'll need a different API (additional pgbin directories where you match the exact dir like ```/opt/homebrew/postgresql/10.1```? just brainstorming), or you'll need to change the impl to make sure we're talking about a pg_bin_dir (check binaries' names?)

Reply:
I installed it with Homebrew on my Mac, here's my locations:
```
$ which psql
/opt/homebrew/bin/psql
$ which pg_dump
/opt/homebrew/bin/pg_dump
```
however... these are actually symlinks:
```
/opt/homebrew/bin/psql -> ../Cellar/postgresql/14.1_1/bin/psql
/opt/homebrew/bin/pg_dump -> ../Cellar/postgresql/14.1_1/bin/pg_dump
```
... which point to real files (i.e. the symlinking ends there):
```
$ ls -l /opt/homebrew/Cellar/postgresql/14.1_1/bin/psql
/opt/homebrew/Cellar/postgresql/14.1_1/bin/psql
$ ls -l /opt/homebrew/Cellar/postgresql/14.1_1/bin/pg_dump
/opt/homebrew/Cellar/postgresql/14.1_1/bin/pg_dump
```

Reply:
@alanfranz-at-work as we have tested two distributions on mac it goes with `/{major}.{minor}/` without a prefix, please let me know if you have a better idea to limit the search scope instead of `(usr_dir, r"([0-9]+(\.[0-9]+)*)")`